builtInFunctions/esdtDataStorage.go

```diff
@@ -369,16 +376,19 @@ func (e *esdtDataStorage) SaveESDTNFTToken(
 		return nil, err
 	}
 
-	tokenTypeChanged, newTokenType, err := e.checkTokenTypeChanged(esdtTokenKey, esdtData)
-	if err != nil {
-		return nil, err
-	}
-	if tokenTypeChanged {
-		esdtData.Type = newTokenType
-		if newTokenType == uint32(core.NonFungibleV2) {
-			err = e.removeMetaDataFromSystemAccount(esdtTokenKey, nonce)
-			if err != nil {
-				return nil, err
+	esdtIdentifier := getEsdtIdentifierWithoutBaseKeyPrefix(esdtTokenKey)
+	if !e.crossChainTokenCheckerHandler.IsCrossChainOperation(esdtIdentifier) {
+		tokenTypeChanged, newTokenType, err := e.checkTokenTypeChanged(esdtTokenKey, esdtData)
```

Comment:
We don't need the type check/change in system account for prefixed tokens.
cc @BeniaminDrasovean @sasurobert

Reply:
Exacly. This is used if an `SFT/metaESDT` is changed to `DynamicSFT/DynamicMetaESDT`, but a transaction to change the type can not be executed from another chain on the main chain, so this will not be needed.